### PR TITLE
feat: Collect usage metrics on license renewal (no-changelog)

### PR DESCRIPTION
**Pull request by csuermann**

### CODE REVIEW

File: packages/cli/src/License.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ import {
 	UNLIMITED_LICENSE_QUOTA,
 } from './constants';
 import Container, { Service } from 'typedi';
+import { WorkflowRepository } from './databases/repositories';
```
Comment (ivov) [inline packages/cli/src/License.ts:15]:
```suggestion
import { WorkflowRepository } from '@/databases/repositories';
```

File: packages/cli/src/License.ts [hunk 2]
```diff
@@ -79,6 +84,15 @@ export class License {
 		}
 	}
 
+	async collectUsageMetrics() {
```
Comment (ivov) [inline packages/cli/src/License.ts:87]: We may want to add number of executions as well?

File: packages/cli/test/unit/License.test.ts [hunk 3]
```diff
@@ -39,6 +39,7 @@ describe('License', () => {
 			loadCertStr: expect.any(Function),
 			saveCertStr: expect.any(Function),
 			onFeatureChange: expect.any(Function),
+			collectUsageMetrics: expect.any(Function),
```
Comment (ivov) [inline packages/cli/test/unit/License.test.ts:42]: Or tighter:

```suggestion
			collectUsageMetrics: expect.any(async () => Promise<Array<{ name: string; value: string }>>),
```

File: packages/cli/src/License.ts [hunk 4]
```diff
@@ -79,6 +84,15 @@ export class License {
...
+		return [
+			{
+				name: 'activeWorkflows',
+				value: await Container.get(WorkflowRepository).count({ where: { active: true } }),
+			},
```
Comment (ivov) [inline packages/cli/src/License.ts:92]: > Products can be configured to support the collection of usage metrics from consumers. `product.metrics` is an array of `{name: string, type: 'counter' | 'gauge'}` objects

Why the difference between `value` and `type`? Are the docs outdated?

**Review by ivov — DISMISSED** [review]
LGTM. Only minor comments.

File: packages/cli/src/License.ts (hunk 2 above)
Comment (csuermann) [reply packages/cli/src/License.ts:87]: this is currently not a requirement.

File: packages/cli/src/License.ts (hunk 4 above)
Comment (csuermann) [reply packages/cli/src/License.ts:92]: `product.metrics` refers to the metric **definitions** maintained in License Server, whereas the SDK's `collectUsageMetrics()` callback expects an array of name / value pairs.

**Review by ivov — DISMISSED** [review]

File: packages/cli/test/unit/License.test.ts (hunk 3 above)
Comment (csuermann) [reply packages/cli/test/unit/License.test.ts:42]: I gave up making this error go away:

> TypeError: Function has non-object prototype 'undefined' in instanceof check

**Review by ivov — DISMISSED** [review]

File: packages/cli/src/License.ts [hunk 5]
```diff
@@ -65,6 +69,7 @@ export class License {
 				loadCertStr: async () => this.loadCertStr(),
 				saveCertStr,
 				deviceFingerprint: () => this.instanceSettings.instanceId,
+				collectUsageMetrics,
```
Comment (netroy) [inline packages/cli/src/License.ts:72]: I'm guessing this isn't going to be called frequently enough to try to optimize the DB calls made here?

**Review by netroy — APPROVED** [review]

File: packages/cli/src/License.ts (hunk 5 above)
Comment (csuermann) [reply packages/cli/src/License.ts:72]: `collectUsageMetrics()` is only called prior license renewal, which happens once every 10 days by default. It is also triggered when n8n starts up and when the _usage and plan_ page is opened, but probably not often enough to optimize DB calls 😄

**Comment by janober** [comment]
Got released with [`n8n@1.14.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.14.0)

**Pull request merged**